packages/web-components/@groupby/elements-base/src/base.ts

```diff
@@ -15,3 +20,30 @@ export default abstract class Base extends LitElement {
     return this.dispatchEvent(eventToDispatch);
   }
 }
+
+/**
+ * A property decorator that will set the data initialization state in a class.
+ * It will extend the property's setter to set the initialization state to true.
+ * This expects that the property already has defined accessors and a default value.
+ *
+ * @param initProperty The name of the initialization property. Has a default value
+ * of "_initialized".
+ */
+export function dataInitializer(initProperty: string = '_initialized'): PropertyDecorator {
+  // eslint-disable-next-line func-names
+  return function(target: object, propertyName: string): void {
+    const oldDescriptor = Object.getOwnPropertyDescriptor(target, propertyName);
+    const instanceMap = new WeakMap();
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    function decoratedSetter(newVal: any): void {
```

Comment:
I personally don't have a problem with this, but most will recommend the following form instead.

```suggestion
    const decoratedSetter = function(newVal: any): void {
```

Reply:
After talking it through we decided to keep it as is.